storage/pipeline/sealing.go

```diff
@@ -304,6 +304,10 @@ func (m *Sealing) TerminateSector(ctx context.Context, sid abi.SectorNumber) err
 	return m.sectors.Send(uint64(sid), SectorTerminate{})
 }
 
+func (m *Sealing) SectorsSummary(ctx context.Context) map[SectorState]int64 {
+	return m.stats.byState
```

Comment:
* You have to copy this map before returning from this function, otherwise the map may be read and modified in parallel which will crash the process
  * You can move the map copy from `StorageMinerAPI.SectorsSummary` into here for that
* When accessing the `m.stats.byState` map in this function you have to hold the `m.stats.lk` lock

Reply:
@magik6k thanks for the review, just patched by moving the map copy to storage/pipeline/sealing.go and add the lock management before the copy with

```
m.stats.lk.Lock()
defer m.stats.lk.Unlock()
```